python/pyspark/sql/pandas/conversion.py

```diff
@@ -336,10 +338,12 @@ def createDataFrame(self, data, schema=None, samplingRatio=None, verifySchema=Tr
                         "has been set to false.\n  %s" % str(e))
                     warnings.warn(msg)
                     raise
-        data = self._convert_from_pandas(data, schema, timezone)
+
+        should_localize = not self._is_timestamp_ntz_preferred()
```

Comment:
nit: we should pass the result of `self._is_timestamp_ntz_preferred()` to `self._create_from_pandas_with_arrow(...)` at line 321 as well to avoid multiple calls?
Otherwise, we can call it in `self._convert_from_pandas(...)` instead?

Reply:
I mean, I'm just wondering why it's called here.
It can be called earlier to reuse the result, or it can be called later in each function.

Reply:
I think I called it here to make `_convert_from_pandas` indenependent from `SparkSession`. `_convert_from_pandas` doesn't have any reference of `self` ... but looks like I overthought too much ... let me call it there.